python/paddle/fluid/layers/nn.py

```diff
@@ -151,38 +153,63 @@ def fc(input,
     dtype = helper.input_dtype()
 
     mul_results = []
-    for input_var, param_attr in helper.iter_inputs_and_params():
-        input_shape = input_var.shape
+    if use_mkldnn:
+        tmp = helper.create_tmp_variable(dtype)
+        input_shape = input.shape
         param_shape = [
             reduce(lambda a, b: a * b, input_shape[num_flatten_dims:], 1)
         ] + [size]
 
         w = helper.create_parameter(
-            attr=param_attr, shape=param_shape, dtype=dtype, is_bias=False)
-        tmp = helper.create_tmp_variable(dtype)
+            attr=helper.param_attr,
+            shape=param_shape,
+            dtype=dtype,
+            is_bias=False)
+        bias_attr = False
+        if bias_attr is not None:
+            bias_attr = True
```

Comment:
for line 168-170, if `bias_attr=False` from parameters, it will be set to `True` in line 170?

Reply:
Oh,  I agree with you. Thank you.